pandapower/groups.py

```diff
@@ -0,0 +1,464 @@
+        key_diff2 = set(ed2.keys()) - set(ed1.keys())
+        idx_diff = [key for key in set(ed2.keys()) - key_diff2 if len(
+            ed1[key].symmetric_difference(ed2[key]))]
+
+        # --- loggings
+        if len(key_diff1) and verbose:
+            logger.info("Missing element keys in elements_dict_to_compare_with: " + str(key_diff1))
+        if len(key_diff2) and verbose:
+            logger.info("Missing element keys in self.elements_dict: " + str(key_diff2))
+        if len(idx_diff) and verbose:
+            logger.info("Differing indices in " + str(idx_diff))
+
+        return not (len(key_diff1) or len(key_diff2) or len(idx_diff))
+
+    def count_elements(self):
+        """ Returns a Series concluding the number of included elements in self.elements_dict """
+        ser = pd.Series([], dtype=int)
+        for elm, idx in self.elements_dict.items():
+            ser.loc[elm] = len(idx)
+        return ser
+
+    def _sum_powers(self, net, formula_character, unit):
+        power = 0.
+        missing_res_idx = list()
+        no_power_column_found = list()
+        no_res_table_found = list()
+        for elm in self.elements_dict.keys():
+            if elm in ["switch", "measurement", "bus", "bus_geodata", "line_geodata"]:
+                continue
+            idx = self.get_index(net, elm)
+            res_elm = "res_" + elm
+            if res_elm not in net.keys():
+                no_res_table_found.append(elm)
+                continue
+            res_idx = net[res_elm].index.intersection(idx)
+            sign = 1 if elm not in ["ext_grid", "gen", "sgen"] else -1
+            if len(res_idx) != len(idx):
+                missing_res_idx.append(elm)
+            col1 = "%s_%s" % (formula_character, unit)
+            col2 = "%sl_%s" % (formula_character, unit)
+            if col1 in net[res_elm].columns:
+                power += sign * net[res_elm][col1].loc[res_idx].sum()
+            elif col2 in net[res_elm].columns:
+                power += sign * net[res_elm][col2].loc[res_idx].sum()
+            else:
+                no_power_column_found.append(elm)
+
+        if len(no_res_table_found):
+            logger.warning(f"Result tables of this elements does not exist: {no_res_table_found}.")
+        if len(missing_res_idx):
+            logger.warning("The resulting power may be wrong since the result tables of these "
+                           f"elements lack of indices: {missing_res_idx}.")
+        if len(no_power_column_found):
+            logger.warning("The resulting power may be wrong since the result tables of these "
+                           f"elements have no requested power column: {no_power_column_found}.")
+        return power
+
+    def res_p_mw(self, net):
+        return self._sum_powers(net, "p", "mw")
+
+    def res_q_mvar(self, net):
+        return self._sum_powers(net, "q", "mvar")
+
+# --- i/o functionality
+
+@to_serializable.register(Group)
```

Comment:
@SimonRubenDrauz please take a look at the I/O, is this okay from your point of view?